web/src/layout/Breadcrumb.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Breadcrumb } from 'antd';
+import PropTypes from 'prop-types';
+
+const mainModules = ['overview', 'project', 'template', 'resource', 'workflow'];
+const operation = ['update', 'add'];
+
+/**
+ * define the route rules
+ * list page => /project
+ * project detail  => /project/:projectId
+ * update project => /project/:projectId/update
+ * add project => /project/add
+ */
+const BreadcrumbComponent = ({ location }) => {
+  const pathSnippets = location.pathname.split('/').filter(i => i);
+  const extraBreadcrumbItems = pathSnippets.map((path, index) => {
+    const url = `/${pathSnippets.slice(0, index + 1).join('/')}`;
+    let text = path;
+    if (_.includes(mainModules, `${path}`)) {
```

Comment:
`includes(mainModules, path)`

Reply:
i use the `lodash` JavaScript utility library

Reply:
You mistaken it, i mean `_.includes(mainModules, path)`, use `path` directly.

Reply:
ok